src/OpenLoco/src/GameCommands/RaiseLowerLand.cpp

```diff
@@ -0,0 +1,485 @@
+            TileManager::setTerrainStyleAsCleared(pos);
+
+            auto clearHeight = TileManager::getHeight(pos).landHeight;
+            TileManager::removeAllWallsOnTile(toTileSpace(pos), clearHeight);
+        }
+
+        // Compute cost of landscape operation
+        auto tile = TileManager::get(pos);
+        auto* surface = tile.surface();
+        auto landObj = ObjectManager::get<LandObject>(surface->terrain());
+        totalCost = totalCost + Economy::getInflationAdjustedCost(landObj->costFactor, landObj->costIndex, 10);
+
+        // If surface is in use as a water(route), and has water, ensure we don't remove it
+        if (surface->hasType6Flag() && surface->water() > 0)
+        {
+            auto water = surface->water() * 2 - kMicroToSmallZStep;
+            auto testHeight = targetBaseZ;
+            if (slopeFlags & 0x1F)
+            {
+                testHeight += kSmallZStep;
+                if (slopeFlags & SurfaceSlope::doubleHeight)
+                {
+                    testHeight += kSmallZStep;
+                }
+            }
+
+            if (water < testHeight)
+            {
+                setErrorText(StringIds::water_channel_currently_needed_by_ships);
+                return FAILURE;
+            }
+        }
+
+        // Bind our local vars to the tile clear function
+        auto clearFunc = [pos, &removedBuildings, flags, &totalCost](TileElement& el) {
+            return TileClearance::tileClearFunction(el, pos, removedBuildings, flags, totalCost);
+        };
+
+        QuarterTile qt(0xF, 0);
+        auto tileHeight = TileManager::getHeight(pos);
+        if (!TileClearance::applyClearAtAllHeights(pos, tileHeight.landHeight / 4, tileHeight.landHeight / 4, qt, clearFunc))
+        {
+            return GameCommands::FAILURE;
+        }
+
+        // Check bridge requirements for track and road elements
+        const auto tileIt = TileManager::get(pos);
+        for (auto& el : tileIt)
+        {
+            if (!(el.type() == ElementType::track || el.type() == ElementType::road))
+            {
+                continue;
+            }
+
+            if (el.isGhost())
+            {
+                continue;
+            }
+
+            auto height = el.baseZ() - targetBaseZ;
+            if (height < 0)
+            {
+                continue;
+            }
+
+            if (el.type() == ElementType::track)
+            {
+                auto* trackEl = el.as<TrackElement>();
+                if (trackEl != nullptr)
+                {
+                    if (trackEl->hasBridge())
+                    {
+                        auto* bridgeObj = ObjectManager::get<BridgeObject>(trackEl->bridge());
+                        if (height > bridgeObj->maxHeight)
+                        {
+                            setErrorText(StringIds::bridge_already_at_maximum_height);
+                            return FAILURE;
+                        }
+                    }
+
```

Comment:
this should be an else so that if it has a bridge and the height is fine then we can continue.